pyuvdata/uvbeam/cst_beam.py

```diff
@@ -220,36 +226,127 @@ def read_cst_beam(
 
         data = np.loadtxt(filename, skiprows=2)
 
-        theta_col = np.where(np.array(column_names) == "theta")[0][0]
-        phi_col = np.where(np.array(column_names) == "phi")[0][0]
+        theta_col = column_names.index("theta")
+        phi_col = column_names.index("phi")
 
-        if "deg" in units[theta_col]:
-            theta_data = np.radians(data[:, theta_col])
-        else:
-            theta_data = data[:, theta_col]
-        if "deg" in units[phi_col]:
-            phi_data = np.radians(data[:, phi_col])
-        else:
-            phi_data = data[:, phi_col]
+        for icol, un in enumerate(units):
+            if "deg" in un:
+                data[:, icol] = np.radians(data[:, icol])
 
-        theta_axis = np.sort(np.unique(theta_data))
-        phi_axis = np.sort(np.unique(phi_data))
-        if not theta_axis.size * phi_axis.size == theta_data.size:
-            raise ValueError("Data does not appear to be on a grid")
+        theta_data = data[:, theta_col]
+        phi_data = data[:, phi_col]
 
-        theta_data = theta_data.reshape((theta_axis.size, phi_axis.size), order="F")
-        phi_data = phi_data.reshape((theta_axis.size, phi_axis.size), order="F")
+        if thetaphi_to_azza_fnc is not None:
+            phi_data, theta_data = thetaphi_to_azza_fnc(theta_data, phi_data)
 
-        if not uvutils._test_array_constant_spacing(theta_axis, self._axis2_array.tols):
+        if theta_data.max() > np.pi * 1.00001 or theta_data.min() < 0:
             raise ValueError(
-                "Data does not appear to be regularly gridded in zenith angle"
+                "The zenith-angle is outside (0, pi): "
+                f"{theta_data.min(), theta_data.max()}. "
+                "Check your thetaphi_to_azza_fnc."
             )
 
-        if not uvutils._test_array_constant_spacing(phi_axis, self._axis1_array.tols):
+        if phi_data.max() > phi_data.min() + 2 * np.pi:
             raise ValueError(
-                "Data does not appear to be regularly gridded in azimuth angle"
+                "The azimuthal angle is outside (0, 2*pi): "
+                f"{phi_data.min(), phi_data.max()}. "
+                "Check your thetaphi_to_azza_fnc."
```

Comment:
I think we need to reword this. It doesn't look like this requires that it is in the interval [0, 2pi], it requires that it only spans 2pi, right?

Reply:
The part that confuses me is `The azimuthal angle is outside (0, 2*pi)`. I read that to say that it is required to be within the interval [0, 2pi].

Reply:
Got it, yup missed this comment originally. I've fixed it now.